### PR TITLE
CLOUDP-181255: use v2 error handling for clusters

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

CLI does very little error handling where we detect specific error type. We needed to use v2 error handling capability with clusters to test if error object is working properly. Additionally, I'm providing fix for missing name validation when a cluster exists.

## Test performed

```
▶ atlas clusters create gcp --provider GCP --region CENTRAL_US
Error: the name already exists. To find the available names, run "atlas cluster list"
```

```
▶ atlas clusters create gcp --provider GCP --region Roscommon 
Error: the region does not exist. To find the available regions, run "atlas cluster availableRegions list"
```

### CODE REVIEW

File: internal/cli/errors.go [hunk 1]
```diff
@@ -27,3 +27,4 @@ var errMissingProjectID = fmt.Errorf(requiredF, flag.ProjectID)
 var ErrMissingOrgID = fmt.Errorf(requiredF, flag.OrgID)
 var ErrFreeClusterAlreadyExists = errors.New("this project already has another free cluster. To learn how to create non-free clusters, run \"atlas cluster create --help\"")
 var ErrNoRegionExistsTryCommand = errors.New(`the region does not exist. To find the available regions, run "atlas cluster availableRegions list"`)
+var ErrNameExists = errors.New(`the name already exists. To find the available names, run "atlas cluster list"`)
```
Comment (gssbzn) [inline internal/cli/errors.go:30]: atlas cluster list will return taken names not available names

**Review by sarahsimpers — COMMENTED** [review]
Copy LGTM

File: internal/cli/atlas/clusters/create.go [hunk 2]
```diff
@@ -81,11 +80,14 @@ func (opts *CreateOpts) Run() error {
 	}
 
 	r, err := opts.store.CreateCluster(cluster)
-	var target *atlas.ErrorResponse
-	if errors.As(err, &target) && target.ErrorCode == "INVALID_ATTRIBUTE" && strings.Contains(target.Detail, "regionName") {
+	apiError, ok := atlasv2.AsError(err);
+	code := apiError.GetErrorCode();
+	if ok && apiError.GetErrorCode() == "INVALID_ATTRIBUTE" && strings.Contains(apiError.GetDetail(), "regionName") {
```
Comment (swattyT) [inline internal/cli/atlas/clusters/create.go:85]: In v2, this is the Error struct:
type Error struct {
	Detail *string `json:"detail,omitempty"`
	// HTTP status code returned with this error.
	Error *int `json:"error,omitempty"`
	// Application error code returned with this error.
	ErrorCode *string `json:"errorCode,omitempty"`
	// Parameter uses to give more information about the error.
	Parameters []interface{} `json:"parameters,omitempty"`
	// Application error message returned with this error.
	Reason *string `json:"reason,omitempty"`
}

Meaning `apiError.GetDetail()` will probably give the HTTPCode based on the comment description.

Comment (wtrocki) [reply internal/cli/atlas/clusters/create.go:85]:
```
{"detail":"Invalid attribute regionName specified.","error":400,"errorCode":"INVALID_ATTRIBUTE","parameters":["regionName"],"reason":"Bad Request"}
```

Comment (swattyT) [reply internal/cli/atlas/clusters/create.go]: Thanks for adding this!

**Review by swattyT — APPROVED** [review]

**Pull request merged**